1649 merge angular and spring upgrades

camerojo opened the pull request.

sadatmalik committed:
update Readme.md with docker compose services details

sadatmalik committed:
add postgres, elasticsearch and kibana services to docker-compose

sadatmalik committed:
update admin-portal from angular 13 to 14. use untyped form controls, builders, and arrays for backward compatibility

sadatmalik committed:
Route.pathMatch is no longer compatible with generic string values - explicitly type string values as const

sadatmalik committed:
remove bootstrap dependency temporarily

sadatmalik committed:
remove ng-select dependency temporarily

sadatmalik committed:
remove ngx-wig dependency temporarily

sadatmalik committed:
update from angular 14 to 15 and to ng-bootstrap v14

sadatmalik committed:
updating to ng-bootstrap v14 for angular 15 compatibility

sadatmalik committed:
updating to ngx-wig v15.1.4 for angular 15 compatibility

sadatmalik committed:
fix scss imports

sadatmalik committed:
add title to all routes - it is expected in angular 15

sadatmalik committed:
fix breaking unit tests

sadatmalik committed:
run upgrade from angular 15 to 16

sadatmalik committed:
upgrade ng-bootstrap, ng-select and ngx-wig to versions compatible with angular 16

sadatmalik committed:
update dragula version for angular 16 compatability

sadatmalik committed:
fix breaking unit test

Commit:
Add SpringBoot version 3.3.5 to gradle

Convert javax to jakarta

Commit:
Deleted org/tctalent/server/configuration/WebSocketSecurityConfig.java and replaced with EnableWebSocketSecurity on TcTalentApplication

Commit:
Replace javax.servlet with jakarta.servlet

Commit:
New DatabaseConfiguration replacing strategy

Commit:
Update to new Thymeleaf

Commit:
Replaced javax.transaction with jakarta

sadatmalik committed:
update ngx-infinite-scroll to angular 16 compatible version

sadatmalik committed:
remove dependency on angular-2-local-storage + update chart.js and ng2-charts to angular 16 compatible version

sadatmalik committed:
create a native local storage wrapper service + unit tests

sadatmalik committed:
remove angular-2-local-storage dependencies with native local storage service wrapper

sadatmalik committed:
update ngs-translate to angular 16 compatible version

sadatmalik committed:
update chart component and template for new charts.js and ng2-charts versions

sadatmalik committed:
fix breaking unit tests - use internal local storage service

sadatmalik committed:
fix breaking unit tests - use updated types for chartDataSet and chartOptions

sadatmalik committed:
fix breaking unit tests - update expected return types

sadatmalik committed:
add error handling to storage service

sadatmalik committed:
use tc-admin for storage prefix

sadatmalik committed:
fix final breaking unit tests - all tests now pass

Commit:
Merge a standard SpringInitializr gradle build

camerojo committed:
Calculate isDev.

Remove unnecessary security dependencies given we have security starter

camerojo committed:
Calculate isDev and set as external property

Remove unnecessary security dependencies given we have security starter

camerojo committed:
Java8TimeDialect is now embedded into the core Thymeleaf - see https://www.thymeleaf.org/doc/articles/thymeleaf31whatsnew.html

camerojo committed:
jakartaActivation is a built in Boot dependency

camerojo committed:
Replace EmailSender with a Spring email version.

sadatmalik committed:
resolve issue with gradle build - ngcc is no longer needed by angular 16, as Ivy is the only rendering engine supported

sadatmalik committed:
resolve gradle build warning - update browserlist to recommended settings

sadatmalik committed:
resolve gradle build error - ngb-highlight directive requires explicit terms attribute

sadatmalik committed:
resolve gradle build warning - angular 16 defaults to es2022

sadatmalik committed:
resolve gradle build error - fix TS inheritance issue by adding `declare` for inherited properties - retested jobs and cases search filters work as expected

sadatmalik committed:
resolve gradle build error - fix TS inheritance issue by removing inherited properties - retested job summary autosave works as expected

sadatmalik committed:
resolve gradle build error - fix TS inheritance issue by explicitly declaring inherited property

sadatmalik committed:
resolve gradle build error - fix TS inheritance issue by explicitly declaring or remove inherited property

sadatmalik committed:
update candidate-portal to angular version 14

sadatmalik committed:
migrate to untyped forms

sadatmalik committed:
temporarily remove bootstrap dependency

sadatmalik committed:
temporarily remove ng-select dependency

sadatmalik committed:
temporarily remove ngx-wig dependency

sadatmalik committed:
apply update from angular 14 to 15

sadatmalik committed:
update bootstrap, ng-select, ngx-wig to  angular 15 compatible versions

sadatmalik committed:
fix scss imports

sadatmalik committed:
adding route titles - expected in angular 15

sadatmalik committed:
update angular version from 15 to 16

sadatmalik committed:
upgrade ng-bootstrap, ng-select and ngx-wig to versions compatible with angular 16

sadatmalik committed:
remove dependency on angular-2-local-storage

sadatmalik committed:
create a native local storage wrapper service

sadatmalik committed:
replace angular-2-local-storage dependencies with native storage service

sadatmalik committed:
update ngs-translate to angular 16 compatible version

sadatmalik committed:
these static properties are undefined at startup and angular 16's stricter DI checks will not allow the app to start - make them instance properties instead

sadatmalik committed:
ensure only a single instance of language loader is created and reused

Commit:
Add in org.testcontainers

Commit:
New ES configuration

Commit:
Update doc of CandidateEsRepository

sadatmalik committed:
resolve gradle build warning - angular 16 defaults to es2022

sadatmalik committed:
Angular DI has stricter initialisation behavior with ES2022 - accessing services directly in property initialisation can result in undefined errors and app startup fails - to resolve: access services in constructor

sadatmalik committed:
update public-portal from angular 13 to 14

sadatmalik committed:
remove bootstrap dependency temporarily

sadatmalik committed:
update from angular 14 to 1

sadatmalik committed:
updating to ng-bootstrap v14 for angular 15 compatibility

sadatmalik committed:
fix scss imports

sadatmalik committed:
run upgrade from angular 15 to 16

sadatmalik committed:
upgrade ng-bootstrap to v15 for compatibility with angular 16

sadatmalik committed:
resolve gradle build warning - angular 16 defaults to es2022

sadatmalik committed:
update readme to install node 18

sadatmalik committed:
add intellij params and update readme - closes #1311

Commit:
Upgrade of Background task support to work with new Spring Trigger

Commit:
Deleted old EmailTester - no longer useful

Commit:
Change gradle build to allow Lombok to be used in testing code

Commit:
Delete old CandidateEsRepositoryTest.java - this will be replaced with something more useful.

Commit:
First cut at converting Elasticsearch code to use new Spring Data implementation

Lots of TODOs in here

camerojo committed:
Add missing security messaging dependency needed by websockets but not included in websocket starter

sadatmalik committed:
fixes broken unit tests - injected service should not be used for direct field initialisation

camerojo committed:
Add arg names to Auditor - was causing error in conversion to SpringBoot 3

Commit:
Can't use Flyway dependency generated by Spring Initializr

camerojo committed:
Add @Qualifier to remove ambiguity about which TemplateEngine to inject.

Spring Boot 3 does not allow this ambiguity.

camerojo committed:
No change - just reordered code

camerojo committed:
Add WebSocketSecurityConfig back in - even though it is a deprecated approach

camerojo committed:
No functional change. Doc only.

camerojo committed:
Doc only

camerojo committed:
Reinstate Postgres CandidateRepository wild card searches.

Reinstate CandidateService calling Postgres wild card searches.

Remove no longer needed ElasticsearchService equivalents (which no longer need to be ported across to new Elasticsearch API.

camerojo committed:
Now need to name the DestinationVariable

camerojo committed:
Remove @EnableWebSocketSecurity to match pre Spring Boot 3 code. Doesn't seem to be necessary here.

camerojo committed:
Compile with -parameters

Exclude commons-logging

camerojo committed:
Get logging of ElasticSearch query working

camerojo committed:
Refactor the actual ES search from SavedSearchServiceImpl to ElasticsearchServiceImpl

camerojo committed:
Move convert query to JSON call to ElasticsearchServiceImpl

camerojo committed:
Fix bug where null values were not handles in range filter.

Do some unit tests

Commit:
Refactored searchCandidateEs to take builder and PageRequest

Commit:
Doc only

camerojo committed:
Merge branch 'refs/heads/staging' into 1614-spring-upgrade-preparation-replace-javax-classes-with-jakarta-classes

camerojo committed:
First go at getting rid of using getExpressions in Specifications

Commit:
Improved model for use of Predicates

Commit:
Improved model for use of Predicates

camerojo committed:
Remove unnecessary cb.or

camerojo committed:
Converted remaining specifications to eliminate Predicate.getExpressions().add(... code

camerojo committed:
Don't add empty disjunctions to conjunctions (because they will always be false)

camerojo committed:
Replace deprecated rawStatusCode method

camerojo committed:
Set redundant Job opp fields

camerojo committed:
Enable Lombok for tests as well

camerojo committed:
Convert a couple of simple deprecations

sadatmalik committed:
merge from staging and resolve conflicts

camerojo committed:
Merge branch 'refs/heads/1403-create-tc-services-in-docker-compose' into 1649-mergeSpringAndAngularUpgrades

camerojo committed:
Fix conversion of Long to BigInteger

camerojo committed:
Fix conversion of Long to BigInteger

Commit:
Disable security on docker-compose.yml ElasticSearch

Commit:
Fix SiteRedirectControllerTest. Have to use .param to specify uri query params

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java:
```diff
@@ -972,7 +958,7 @@ private Specification<Candidate> computeQuery(
     }
 
-    private BoolQueryBuilder computeElasticQuery(
+    private BoolQuery.Builder computeElasticQuery(
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1130-1273 into a separate method.

sadatmalik replied:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1135-1285 into a separate method.

sadatmalik replied:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1144-1294 into a separate method.

camerojo committed:
Disable Elasticsearch SpringBootTest in build.gradle

camerojo committed:
Add the "skip-test-in-gradle-build" tag (renamed from "spring-boot-test") to all SpringBootTests

camerojo committed:
Support for subqueries.

camerojo committed:
Get nested queries going

Commit:
Change addElasticNestedFilter to take a builder instead of the nested query.

Deleted the scoreMode - not needed.

Commit:
Add optional SearchType for addElasticBooleanFilter

Better tests for Nested filters.

Get Occupations and Language proficiency tests working.

camerojo commented on README.md:
Great doc! I love how it is getting simpler for new developers.

camerojo reviewed (commented):
Labour of Hercules! Feel great to have it in.
